client.go

```diff
@@ -368,20 +368,43 @@ func prepareMultipartData(request *Request) (*prepareMultipartDataResult, error)
 	if err != nil {
 		return nil, err
 	}
-	fw.Write(input)
+	_, err = fw.Write(input)
+	if err != nil {
+		return nil, err
+	}
 	fw, err = mpw.CreateFormField("map")
 	if err != nil {
 		return nil, err
 	}
+	_, err = fw.Write([]byte("{"))
```

Comment:
instead of manually writing the json here you could use:
```golang
json.NewEncoder(fw).Encode(res.fileMap)
```
to write the already existing map to the stream. The slice in the value is only ever of length one as it's never added to.